### DISCUSSION TITLE
Should I update the other side's collection in a OneToMany relationship setter?

### DISCUSSION THREAD

**Discussion by michnovka** [doctrine/doctrine2#10419] [Support Questions]
I have 2 entities in a relationship like this:

```php
#[ORM\Entity]
class Duel
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private int $id;

    #[ORM\OneToMany(mappedBy: 'duel', targetEntity: Game::class)]
    private Collection $games;

    public function getGames(): Collection
    {
        return $this->games;
    }

    public function addGame(Game $game): self
    {
        if (!$this->games->contains($game)) {
            $this->games->add($game);
            $game->setDuel($this);
        }

        return $this;
    }

    public function removeGame(Game $game): self
    {
        $this->games->removeElement($game);

        return $this;
    }
}
```

and

```php

#[ORM\Entity]
class Game
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private int $id;

    #[ORM\ManyToOne(inversedBy: 'games')]
    private Duel $duel;

    public function getDuel(): Duel
    {
        return $this->duel;
    }

    public function setDuel(Duel $duel): self
    {
        $this->duel = $duel;

        return $this;
    }
```

My question is, should I update the `setDuel` function to this:

```php
    public function setDuel(Duel $duel): self
    {
        $this->duel = $duel;

        $this->duel->addGame($this);
        
        return $this;
    }
```

The reason is that in my code, I was creating new `Game` object and persisting it like this (all part of a transaction)

```php
$game = new Game();
$game->setDuel($duel);
$this->entityManager->persist($game);
$this->entityManager->flush();
```

and later, before the transaction was committed, I attempted to retrieve the duels

```php
// ... some other part of code
$allDuelGames = $duel->getGames();
```
and of course, the `Games` `Collection` was empty.

I had to change `$game->setDuel($duel);` to  `$duel->addGame($game);`, which correctly sets both sides of the relationship on both the `Game` and the `Duel`.

So by the same logic, should my setter on `Game` also update the relationship on `Duel` instance?

The symfony maker bundle does generate correctly updating collections on the `OneToMany` side, but not on the `ManyToOne`, is there a reason for this?

Thanks!

**Comment by mpdude** [comment]
I don’t know anything about the Maker bundle.

But, if you want both sides of the association to be in sync, you have to keep them in sync.

if you can make sure that all new entities will always be added on the same side of the association only, you only need to sync from there.

If you can do both, set the duel on the game and add the game to the duel, you need both.

This is also the reason why you should really think about which associations and directions are necessary, so you don’t code what you don’t need.
